subsys/mpsl/fem/nrf21540_gpio/mpsl_fem_nrf21540_gpio.c

```diff
@@ -4,26 +4,41 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+#if defined(CONFIG_MPSL_FEM_NRF21540_GPIO)
+
 #include <zephyr/devicetree.h>
 #include <zephyr/drivers/gpio.h>
 #include <string.h>
 #include <zephyr/sys/__assert.h>
+
+#if !defined(CONFIG_MPSL_FEM_PIN_FORWARDER)
 #include <mpsl_fem_config_nrf21540_gpio.h>
-#include <mpsl_fem_config_simple_gpio.h>
 #include <nrfx_gpiote.h>
 #if IS_ENABLED(CONFIG_HAS_HW_NRF_PPI)
 #include <nrfx_ppi.h>
 #elif IS_ENABLED(CONFIG_HAS_HW_NRF_DPPIC)
 #include <nrfx_dppi.h>
 #endif
+#else /* !defined(CONFIG_MPSL_FEM_PIN_FORWARDER) */
+#include <soc_secure.h>
+#if IS_ENABLED(CONFIG_PINCTRL)
+#include <pinctrl_soc.h>
+#endif
+#endif /* !defined(CONFIG_MPSL_FEM_PIN_FORWARDER) */
+
+#if !DT_NODE_EXISTS(DT_NODELABEL(nrf_radio_fem))
+#error Node with label 'nrf_radio_fem' not found in the devicetree.
+#endif
+
+#define MPSL_FEM_SPI_IF DT_PHANDLE(DT_NODELABEL(nrf_radio_fem), spi_if)
```

Comment:
Do we want to reuse the code for the gpio and gpio_spi fems somehow together or keep them entirely in separate files (duplicating large amounts of the code for flexibility)?. If the latter then this should be deleted, as SPI has no usage for a GPIO-only FEM.

Reply:
So far we've allowed the user to specify nRF21540's SPI interface in devicetree, but choose the GPIO variant anyway. It's convenient when an existing devicetree file that provides a complete description of nRF21540 (e.g. https://github.com/nrfconnect/sdk-nrf/blob/main/boards/shields/nrf21540_ek/nrf21540_ek.overlay) is applied directly.

Now the question is whether we expect the MPSL FEM subsystem to set those pins to some specific (or at least well-defined state). If we do, then we need to keep all this boilerplate, forward the pins from the application core to the network core and let the network core handle it. Otherwise we can remove it from this variant.

Probably pinctrl sets those SPI pins to a default state. Then we could remove this code. I need to check

Reply:
I removed the code that forwarded SPI pins to the network core - they should be taken care of by pinctrl.